src/output/workspace-impl.cpp

```diff
@@ -783,6 +783,11 @@ class output_viewport_manager_t
 
         for (auto& v : fixed_views)
         {
+            wf::view_change_viewport_signal vdata;
+            vdata.view = v;
+            vdata.from = data.old_viewport;
```

Comment:
Fixed views are not necessarily on the current workspace. You can have fixed views everywhere (not that I know of any plugin which does that but still)

Reply:
Ok so to do this I used `get_view_main_workspace()` which meant that I have to save the old workspaces now since that function corrects for invalid workspace in a lossy way. Is this what you had in mind?

Reply:
Hmm, I haven't thought it in so much details but what you did makes sense, we need to somehow keep this mapping.